addons/binding/org.openhab.binding.mcp23017/ESH-INF/thing/thing-types.xml

```diff
@@ -44,17 +76,47 @@
                 <description>I2C Bus mcp2317 device Address</description>
                 <default>20</default>
             </parameter>
+            <parameter name="bus_number" type="integer" required="true">
+                <label>I2C BUS Number</label>
+                <description>I2C Bus number (0,1)</description>
+                <options>
+                    <option value="0">0</option>
+                    <option value="1">1</option>
+                </options>
+                <default>1</default>
+            </parameter>
             <parameter name="pin" type="text" required="true">
                 <label>PIN name</label>
                 <description>mcp2317 pin name</description>
+                <options>
+                    <option value="A0">A0</option>
+                    <option value="A1">A1</option>
+                    <option value="A2">A2</option>
+                    <option value="A3">A3</option>
+                    <option value="A4">A4</option>
+                    <option value="A5">A5</option>
+                    <option value="A6">A6</option>
+                    <option value="A7">A7</option>
+                    <option value="B0">B0</option>
+                    <option value="B1">B1</option>
+                    <option value="B2">B2</option>
+                    <option value="B3">B3</option>
+                    <option value="B4">B4</option>
+                    <option value="B5">B5</option>
+                    <option value="B6">B6</option>
+                    <option value="B7">B7</option>
+                </options>
                 <default>B0</default>
             </parameter>
             <parameter name="pull_mode" type="text" required="true">
                 <label>PullResistor mode</label>
                 <description>mcp2317 input pull resistor mode</description>
-                <default>PULL_DOWN</default>
+                <options>
+                    <option value="OFF">OFF</option>
+                    <option value="PULL_UP">Internal PULL_UP</option>
+                    <option value="PULL_DOWN">External PULL_DOWN</option>
```

Comment:
@falkena, You are right, I also checked pi4j implementation and PULL_DOWN has no effect. I changed a code, so now is only PULL_UP and OFF, available, so all fixes for Your review done... :)